hwbench/environment/block_devices.py

```diff
@@ -0,0 +1,130 @@
+import pathlib
+from json import JSONDecodeError, loads
+
+import pyudev
+
+from hwbench.utils import helpers as h
+from hwbench.utils.external import External
+
+
+class Block_Device:
+    """Block_Device is a class that gathers block_device information"""
+
+    def __init__(self, out_dir, udev_device):
+        self.udev_device = udev_device
+        self.name = udev_device.sys_name
+        self.type = self.get_block_device_type()
+        self.out_dir = out_dir
+        self.smart_json = b"{}"
+
+    def is_rotational(self) -> bool:
+        syspath = f"/sys/block/{self.udev_device.sys_name}/queue/rotational"
+        with open(syspath) as file:
+            content = file.read()
+        return bool(content.strip())
+
+    def is_ata(self) -> bool:
+        bus = self.udev_device.get("ID_BUS")
+        return bool(bus == "ata")
+
+    def is_nvme(self) -> bool:
+        return bool(self.udev_device.parent.get("NVME_TRTYPE"))
+
+    def get_block_device_type(self):
```

Comment:
if we give a block device that is not an ATA one, the return type can be undefined.
Please add a default return value.